controllers/dataplane/openstackdataplanenodeset_controller.go

```diff
@@ -514,7 +519,12 @@ func checkDeployment(ctx context.Context, helper *helper.Helper,
 			} else if deploymentConditions.IsFalse(dataplanev1.NodeSetDeploymentReadyCondition) {
 				isDeploymentRunning = true
 			} else if deploymentConditions.IsTrue(dataplanev1.NodeSetDeploymentReadyCondition) {
-				if deployment.Status.NodeSetHashes[instance.Name] != instance.Status.ConfigHash {
+				// If the nodeset configHash does not match with what's in the deployment and the
```

Comment:
Like here for example?

```
	if instance.Generation != instance.Status.ObservedGeneration {
 		generationChanged = true
 	}
```

Reply:
we set `instance.Status.ObservedGeneration` to `instance.Generation` at the beginning of the reconcile for a reason, so `ObservedGeneration` would be always the same as `instance.Generation` here and it won't work.